src/sparse_blas/sparse_blas_loader.cpp

```diff
@@ -0,0 +1,240 @@
+DEFINE_SET_CSR_DATA(std::complex<float>, _cf, std::int64_t, _i64)
+DEFINE_SET_CSR_DATA(std::complex<double>, _cd, std::int64_t, _i64)
+#undef DEFINE_SET_CSR_DATA
+
+sycl::event optimize_gemv(sycl::queue &queue, transpose transpose_val, matrix_handle_t handle,
+                          const std::vector<sycl::event> &dependencies) {
+    auto libkey = get_device_id(queue);
+    return function_tables[libkey].optimize_gemv(queue, transpose_val, handle, dependencies);
+}
+
+sycl::event optimize_trmv(sycl::queue &queue, uplo uplo_val, transpose transpose_val, diag diag_val,
+                          matrix_handle_t handle, const std::vector<sycl::event> &dependencies) {
+    auto libkey = get_device_id(queue);
+    return function_tables[libkey].optimize_trmv(queue, uplo_val, transpose_val, diag_val, handle,
+                                                 dependencies);
+}
+
+sycl::event optimize_trsv(sycl::queue &queue, uplo uplo_val, transpose transpose_val, diag diag_val,
+                          matrix_handle_t handle, const std::vector<sycl::event> &dependencies) {
+    auto libkey = get_device_id(queue);
+    return function_tables[libkey].optimize_trsv(queue, uplo_val, transpose_val, diag_val, handle,
+                                                 dependencies);
+}
+
+#define DEFINE_GEMV(FP_TYPE, FP_SUFFIX)                                                           \
+    template <>                                                                                   \
+    void gemv(sycl::queue &queue, transpose transpose_val, const FP_TYPE alpha,                   \
+              matrix_handle_t A_handle, sycl::buffer<FP_TYPE, 1> &x, const FP_TYPE beta,          \
+              sycl::buffer<FP_TYPE, 1> &y) {                                                      \
+        auto libkey = get_device_id(queue);                                                       \
+        function_tables[libkey].gemv_buffer##FP_SUFFIX(queue, transpose_val, alpha, A_handle, x,  \
+                                                       beta, y);                                  \
+    }                                                                                             \
+    template <>                                                                                   \
+    sycl::event gemv(sycl::queue &queue, transpose transpose_val, const FP_TYPE alpha,            \
+                     matrix_handle_t A_handle, const FP_TYPE *x, const FP_TYPE beta,              \
+                     const FP_TYPE *y, const std::vector<sycl::event> &dependencies) {            \
+        auto libkey = get_device_id(queue);                                                       \
+        return function_tables[libkey].gemv_usm##FP_SUFFIX(queue, transpose_val, alpha, A_handle, \
+                                                           x, beta, y, dependencies);             \
+    }
+
+DEFINE_GEMV(float, _sf)
+DEFINE_GEMV(double, _sd)
```

Comment:
What's the `s` stand for? I would've thought we'd be using `r` here for real data type and `c` for complex?

Reply:
I would prefer `r` to align with oneMKL product and with the mathematical intent. @spencerpatty, @noffermans, if you have a preference here, please let us know.

Reply:
For integer data type we can just use `_i32` or similar suffixes with the assumption that we will never have complex integer type. These suffixes are internal only so can be easily changed.
I'll change `s` to `r` here.

Reply:
The suffixes are renamed as of https://github.com/oneapi-src/oneMKL/pull/374/commits/3228b12622b8feb6caa109952d825e0cd65af171